odoo/addons/base/models/ir_module.py

```diff
@@ -919,6 +919,19 @@ def _check(self):
             if not module.description_html:
                 _logger.warning('module %s: description is empty !', module.name)
 
+    def _get(self, name):
+        """ Return the (sudoed) `ir.module.module` record with the given name.
+        The result may be an empty recordset if the module is not found.
+        """
+        model_id = self._get_id(name) if name else False
+        return self.sudo().browse(model_id)
```

Comment:
I'd put the `sudo()` call in the tail position, I think that'd express the intent better.